peer/brontide.go

```diff
@@ -2105,52 +2115,94 @@ func (p *Brontide) queueHandler() {
 	lazyMsgs := list.New()
 
 	for {
-		// Examine the front of the priority queue, if it is empty check
-		// the low priority queue.
+		// Process the priority queue first. We will not accept any new
+		// messages until this queue is empty.
 		elem := priorityMsgs.Front()
-		if elem == nil {
-			elem = lazyMsgs.Front()
-		}
 
+		// Found an item, we will write it to lnwire.
 		if elem != nil {
-			front := elem.Value.(outgoingMsg)
+			front, ok := elem.Value.(outgoingMsg)
+			if !ok {
+				p.log.Errorf("Unknown message type found in "+
+					"priority queue: %v", elem)
+			}
+
+			p.log.Tracef("Sending priority msg to sendQueue: %s",
+				spew.Sdump(front))
 
-			// There's an element on the queue, try adding
-			// it to the sendQueue. We also watch for
-			// messages on the outgoingQueue, in case the
-			// writeHandler cannot accept messages on the
-			// sendQueue.
+			// There's an element on the queue, try adding it to
+			// the `sendQueue`.
 			select {
 			case p.sendQueue <- front:
-				if front.priority {
-					priorityMsgs.Remove(elem)
-				} else {
-					lazyMsgs.Remove(elem)
-				}
-			case msg := <-p.outgoingQueue:
-				if msg.priority {
-					priorityMsgs.PushBack(msg)
-				} else {
-					lazyMsgs.PushBack(msg)
-				}
+				// We know the elem must from priority queue.
+				priorityMsgs.Remove(elem)
+
+				// Once it's sent, we'll continue to the next
+				// iteration. This means we will try to send
+				// messages again if the queue is not empty,
+				// and we won't accept new messages until it's
+				// empty.
+				continue
+
 			case <-p.quit:
 				return
 			}
-		} else {
-			// If there weren't any messages to send to the
-			// writeHandler, then we'll accept a new message
-			// into the queue from outside sub-systems.
+		}
+
+		// If there are no messages in priority queue, we'll read the
+		// lazy queue or accept new messages into the queue from
+		// outside subsystems.
+		elem = lazyMsgs.Front()
+
+		// If there are no messages in the lazy queue, we'll block
+		// until a new message is received.
+		if elem == nil {
```

Comment:
Why are we first checking if `elem` is `nil` in order to block for any message, and then handle the lazy message outside of an `if` block?

Unless there is an algorithmic reason not to, I believe it would be more consistent/readable to first check if `elem != nil` and proceed with the lazy sending as written below, but this time inside of the `if` block. At the end there could be the generic blocking code that waits for any message as the loop's last resort, outside of any `if` section.

Just personal opinion.

Reply:
So the flow would be:

1. Do a priority-send, if priority queue not empty
2. Do a lazy-send, if lazy queue not empty
3. Wait for any message, if both queues are empty

Reply:
Since lazy-send also waits for new messages there wouldn't be any difference in the overall behavior of the loop.

Reply:
so you mean changing the `if elem == nil` to `if elem != nil`?

> Why are we first checking if elem is nil in order to block for any message, and then handle the lazy message outside of an if block?

This is actually more for readability. The code space is less wrapped this way since we need to check more using `if elem != nil`.

Reply:
> so you mean changing the if elem == nil to if elem != nil?

yes, and executing the lazy-send inside the if block

> This is actually more for readability. The code space is less wrapped this way since we need to check more using `if elem != nil`.

Ok